staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset.go

```diff
@@ -74,6 +75,36 @@ func (a ApplySet) ID() string {
 	return "placeholder-todo"
 }
 
+func (a *ApplySet) LabelsForMember() map[string]string {
+	return map[string]string{
+		"applyset.k8s.io/part-of": a.ID(),
+	}
+}
+
+// addLabels sets our tracking labels on each object; this should be called as part of loading the objects.
+func (a *ApplySet) addLabels(objects []*resource.Info) error {
+	applysetLabels := a.LabelsForMember()
+	for _, obj := range objects {
+		accessor, err := meta.Accessor(obj.Object)
+		if err != nil {
+			return fmt.Errorf("getting accessor: %w", err)
+		}
+		labels := accessor.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
+		}
+		for k, v := range applysetLabels {
+			if _, found := labels[k]; found {
+				return fmt.Errorf("applyset label %q already set in input data", k)
```

Comment:
I noticed that your alternative implementation allows an existing value as long as it matches what we want to apply. Should we do that here too, or stick to this more strict check?